quilt/tools/build.py

```diff
@@ -198,16 +198,20 @@ def build_package(username, package, yaml_path, checks_path=None, dry_run=False,
     """
     def find(key, value):
         """find all nodes transitively"""
-        for k, v in iteritems(value):
-            if k == key:
-                yield v
-            elif isinstance(v, dict):
-                for result in find(key, v):
-                    yield result
-            elif isinstance(v, list):
-                for item in v:
-                    for result in find(key, item):
+        try:
+            vals = iteritems(value)
+            for k, v in vals:
+                if k == key:
+                    yield v
+                elif isinstance(v, dict):
+                    for result in find(key, v):
                         yield result
+                elif isinstance(v, list):
+                    for item in v:
+                        for result in find(key, item):
+                            yield result
+        except AttributeError:
```

Comment:
This feels hacky, and can hide actual errors. Check if the object is an instance of `collections.Iterable`?